packages/dds/tree/src/simple-tree/arrayNodeTypes.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { TreeArrayNode } from "./arrayNode.js";
+import type {
+	ImplicitAllowedTypes,
+	InsertableTreeNodeFromImplicitAllowedTypes,
+} from "./schemaTypes.js";
+import {
+	NodeKind,
+	type TreeNodeSchemaClass,
+	type TreeNodeSchema,
+	type TreeNodeSchemaNonClass,
+	type WithType,
+} from "./core/index.js";
+
+import type { SimpleArrayNodeSchema } from "./simpleSchema.js";
+
+/**
+ * A schema for customizable {@link (TreeArrayNode:interface)}s.
+ * @sealed
+ * @alpha
+ */
+export interface ArrayNodeCustomizableSchema<
+	out TName extends string = string,
+	in out T extends ImplicitAllowedTypes = ImplicitAllowedTypes,
+	out ImplicitlyConstructable extends boolean = true,
+	out TCustomMetadata = unknown,
+> extends TreeNodeSchemaClass<
+			TName,
+			NodeKind.Array,
+			TreeArrayNode<T> & WithType<TName, NodeKind.Array, T>,
+			Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>,
+			ImplicitlyConstructable,
+			T,
+			undefined,
+			TCustomMetadata
+		>,
+		SimpleArrayNodeSchema<TCustomMetadata> {}
+
+/**
+ * A schema for POJO emulation mode {@link (TreeArrayNode:interface)}s.
+ * @sealed
+ * @alpha
+ */
+export interface ArrayNodePojoEmulationSchema<
+	out TName extends string = string,
+	in out T extends ImplicitAllowedTypes = ImplicitAllowedTypes,
+	out ImplicitlyConstructable extends boolean = true,
+	out TCustomMetadata = unknown,
+> extends TreeNodeSchemaNonClass<
+			TName,
+			NodeKind.Array,
+			TreeArrayNode<T> & WithType<TName, NodeKind.Array, T>,
+			Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>,
+			ImplicitlyConstructable,
+			T,
+			undefined,
+			TCustomMetadata
+		>,
+		SimpleArrayNodeSchema<TCustomMetadata> {}
+
+/**
+ * @alpha
```

Comment:
Semantic docs?

Reply:
Oh, I missed that unlike with object, this actually has a distinct name from the schema since we have two schema so it needs its own docs. Makes me think maybe we want a unified type and make the other two system.

Reply:
This now shares it name (and thus docs) with ArrayNodeSchema, a type which has docs.